src/vs/workbench/services/decorations/browser/decorations.ts

```diff
@@ -9,13 +9,16 @@ import { Event } from 'vs/base/common/event';
 import { ColorIdentifier } from 'vs/platform/theme/common/colorRegistry';
 import { IDisposable } from 'vs/base/common/lifecycle';
 import { CancellationToken } from 'vs/base/common/cancellation';
+import { ThemeIcon } from 'vs/platform/theme/common/themeService';
 
 export const IDecorationsService = createDecorator<IDecorationsService>('IFileDecorationsService');
 
 export interface IDecorationData {
 	readonly weight?: number;
 	readonly color?: ColorIdentifier;
 	readonly letter?: string;
+	/** The icon to show, letter is ignored if this is specified */
+	readonly icon?: ThemeIcon;
```

Comment:
Isn't an or-type the better solution, like `readonly letter?: string | ThemeIcon`

Reply:
I did that first and changed because:

- We then need to narrow that type with typeof or something which wasnt as nice internally than just checking if icon was truthy
- letter isn't the greatest description if it could also be an icon
- I figured you may want to allow both icons and letters at one point

Up to you.

Reply:
> We then need to narrow that type with typeof or something which wasnt as nice internally than just checking if icon was truthy

We have [isThemeIcon](https://github.com/microsoft/vscode/blob/082af46e6657da59c6df5a15eb40de2afb79cd46/src/vs/platform/theme/common/themeService.ts#L39-L39) for that. Could/should work

> I figured you may want to allow both icons and letters at one point

That will be tricky because you'll need two fonts: codicon and the letter font...

I am easy, if you wanna merge both now that would best but I can also live without that and tackle it has debt. At one point I want to explore CSS variables for the decoration service

Reply:
Oh great, I'll make the `isThemeIcon` change